### ISSUE TITLE
[BUG] Using different distributed strategies of Megatron-LM to train the llama3.1-8B model results in inconsistent training loss

### ISSUE THREAD

**Issue by cailun01** [NVIDIA/Megatron-LM#1324]
**Describe the bug**
I used 4 distributed training strategies to train llama-3.1-8B from scratch. And I use same dataset and super parameters. But I got inconsistent training losses. 

**To Reproduce**
Steps to reproduce the behavior. The easier it is to reproduce the faster it will get maintainer attention.

**Expected behavior**
Regardless of the distributed training strategy used, the training loss should remain consistent.

**Stack trace/logs**
![Image](https://github.com/user-attachments/assets/fe787e5b-84ce-4038-9e3b-30f8b48d3602)

**Environment (please complete the following information):**

 
| item        | version               |
|--------------|------------------------|
| GPU model         | 8*H20                  |
| GPU driver   | 535.183.06             |
| torch        | 2.4.0+cu124            |
| nccl         | 2.20.5+cuda12.4        |
| megatron-core| 0.9.0 (commit: 1afee59)  |
| flash-attn   | 2.5.7                  |
| TransformerEngine| 1.12.0+7f2afaa        |
| apex         | commit: 2d8302         |

**Proposed fix**
If you have a proposal for how to fix the issue state it here or link to a PR.

**Additional context**

**Comment by sbhavani** [comment]
Not enough info. Need the 4 configurations (TP/PP/DP), actual loss values, and training scripts. Closing for new year cleanup, please reopen with details if needed.
